### DISCUSSION TITLE
How to position Menu component?

### DISCUSSION THREAD

**Discussion by Seanmclem** [ariakit/ariakit#1179] [Questions] [answered]
When using a Menu component, how do you set the position? What is it relative to? I'd love a little bit of jsdoc on the component file, if not in the Readme. Or just a little help with this. Thanks

**Comment by diegohaz** [comment, marked as answer]
```js
useMenuState({ placement: "right" });
```

**Reply by Seanmclem** [reply]
Right of what? How does it know the x/y position?

**Reply by Seanmclem** [reply]
Yeah this context menu integration was exactly what I was looking for  https://twitter.com/diegohaz/status/1510671321977532416?s=21&t=0Zj5EfCGvkFsBZPn-XN32Q

**Reply by diegohaz** [reply]
It wasn't clear if you were referring to v1 or v2. My answer applies to both. The snippet on Twitter only applies to v2. The Menu component is usually positioned relative to the MenuButton component.
